api/v1alpha1/common_types.go

```diff
@@ -14,6 +15,24 @@ type (
 
 	// ConditionReason is a static/programmatic representation of the cause of a status condition.
 	ConditionReason string
+
+	// +k8s:deepcopy-gen=false
```

Comment:
Is this related to the `ScheduleSpecInterface`? Does it need the extra line break, or could it be defined below the `// ScheduleSpecInterface represents [...]` comment? (Same for the same situation below.)

Reply:
Yes, otherwise there would be DeepCopy methods generated for those as well, but we don't need that.
the extra line break is to make sure the `// +k8s:deepcopy-gen=false` comment doesn't get part of the function godoc, otherwise it's part of the godoc documentation which I find annoying/irrelevant (same if it's below the type comment).